sdk/inc/azure/iot/az_iot_adu_client.h

```diff
@@ -0,0 +1,620 @@
+   * Example: 700
+   */
+  int32_t result_code;
+  /**
+   * A code that contains additional information about the result.
+   * Example: 0x80004006
+   */
+  int32_t extended_result_code;
+  /**
+   * Customer-defined free form string to provide additional result details.
+   */
+  az_span result_details;
+  /**
+   * Number of items in \p step_results.
+   */
+  int32_t step_results_count;
+  /**
+   * The results for each step in the update manifest instructions.
+   * The number of steps MUST match the number of steps in the
+   * update manifest for the resulting state to be property generated.
+   */
+  az_iot_adu_client_step_result step_results[AZ_IOT_ADU_CLIENT_MAX_INSTRUCTIONS_STEPS];
+} az_iot_adu_client_install_result;
+
+/**
+ * @brief A set of values that indicate which deployment the agent is currently working on.
+ *
+ */
+typedef struct
+{
+  /**
+   * An integer that corresponds to an action the agent should perform.
+   * @remark Refer to the following defines for the expected values:
+   *         AZ_IOT_ADU_CLIENT_SERVICE_ACTION_APPLY_DEPLOYMENT
+   *         AZ_IOT_ADU_CLIENT_SERVICE_ACTION_CANCEL
+   */
+  int32_t action;
+  /**
+   * ID of current deployment.
+   */
+  az_span id;
+  /**
+   * Time of last deployment retry.
+   */
+  az_span retry_timestamp;
+} az_iot_adu_client_workflow;
+
+/**
+ * @brief A map of file ID to download url.
+ */
+typedef struct
+{
+  /**
+   * File ID, mapped in the updated manifest.
+   */
+  az_span id;
+  /**
+   * Complete url to a file.
+   */
+  az_span url;
+} az_iot_adu_client_file_url;
+
+/**
+ * @brief Structure that holds the parsed contents of the ADU
+ *        request in the Plug and Play writable properties sent
+ *        by the ADU service.
+ */
+typedef struct
+{
+  /**
+   * A set of values that indicate which deployment the agent is currently working on.
+   */
+  az_iot_adu_client_workflow workflow;
+  /**
+   * Description of the content of an update.
+   * @note This will come as an escaped string. The user must unescape it using
+   * an API such as az_json_string_unescape() before subsequently calling
+   * az_iot_adu_client_parse_update_manifest() with it.
+   */
+  az_span update_manifest;
+  /**
+   * A JSON Web Signature (JWS) with JSON Web Keys used for source verification.
+   */
+  az_span update_manifest_signature;
+  /**
+   * Tells the agent which files to download and the hash to use to verify that the files
+   * were downloaded correctly.
+   */
+  az_iot_adu_client_file_url file_urls[AZ_IOT_ADU_CLIENT_MAX_FILE_URL_COUNT];
+  /**
+   * Number of items in \p file_urls.
+   */
+  uint32_t file_urls_count;
+} az_iot_adu_client_update_request;
+
+/**
+ * @brief User-defined properties for handling an update request.
+ *
+ */
+typedef struct
+{
+  /**
+   * The installed criteria which defines a successful installation.
+   */
+  az_span installed_criteria;
+} az_iot_adu_client_update_manifest_instructions_step_handler_properties;
```

Comment:
Can this be removed and directly define `handler_properties` as an az_span below?

Reply:
![image](https://user-images.githubusercontent.com/36710865/199548204-3926e243-c260-458d-9159-7fc3caf162d4.png)

As is formatted in the JSON, this is an object which could theoretically have more values. We attempt to model our structs based on the JSON structure.